src/derivative_operators/multi_dim_bc_operators.jl

```diff
@@ -1,8 +1,26 @@
 
 abstract type MultiDimensionalBC{T, N} <: AbstractBC{T} end
 
+struct MultiDimDirectionalBC{T<:Real,S, D, N} <: MultiDimensionalBC{T, N}
+    stls :: S
+    l :: T
+    r :: T
+end
+
+struct ComposedMultiDimBC{T, S, N, M} <: MultiDimensionalBC{T, N}
+    stencils::S
+end
+
+struct MultiDimBC{N} end
+
+MultiDimBC{dim}(N::Int, approx_order::Int, dx::T) where {dim,T} = finite_stencil(approx_order,dx,dim,N)
```

Comment:
what's N?

Reply:
Overall Dimension of the space in which its operating. I think the declaration in the struct `MultiDimBC{N}` is misleading.
Will replace there with `dim`.